src/js/common/schemaform/save-in-progress/RoutedSavableReviewPage.jsx

```diff
@@ -86,56 +191,136 @@ class RoutedSavableReviewPage extends React.Component {
+            ))}
+          </div>
+        </div>
+        <p><strong>Note:</strong> According to federal law, there are criminal penalties, including a fine and/or imprisonment for up to 5 years, for withholding information or for providing incorrect information. (See 18 U.S.C. 1001)</p>
+        <PrivacyAgreement
+          required
+          onChange={this.props.setPrivacyAgreement}
+          checked={form.data.privacyAgreementAccepted}
+          showError={form.submission.hasAttemptedSubmit}/>
+        <SubmitButtons
+          onBack={this.goBack}
+          onSubmit={this.handleSubmit}
+          submission={form.submission}
+          renderErrorMessage={this.renderErrorMessage}/>
+        {!disableSave && form.submission.status === 'error' ? null : <div>
+          <SaveStatus
+            isLoggedIn={user.login.currentlyLoggedIn}
+            showLoginModal={user.login.showModal}
+            toggleLoginModal={this.props.toggleLoginModal}
+            form={form}>
+          </SaveStatus>
+          <SaveFormLink
+            locationPathname={location.pathname}
+            form={form}
+            user={user}
+            saveAndRedirectToReturnUrl={this.props.saveAndRedirectToReturnUrl}
+            toggleLoginModal={this.props.toggleLoginModal}/>
+        </div>
+        }
       </div>
     );
-
-    return (
-      <ReviewPage
-        {...this.props}
-        setData={this.setData}
-        formContext={getFormContext({ user, form })}
-        contentAfterButtons={form.submission.status === 'error' ? null : contentAfterButtons}
-        renderErrorMessage={this.renderErrorMessage}/>
-    );
   }
 }
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
+  // from ownprops
+  const formConfig = ownProps.route.formConfig;
+  const pageList = ownProps.route.pageList;
+
+  // from redux state
+  const form = state.form;
+  const formData = state.form.data;
+  const openChapters = getReviewPageOpenChapters(state);
+  const user = state.user;
+  const viewedPages = getViewedPages(state);
+
+  const chapterNames = getActiveChapters(formConfig, formData);
+  const disableSave = formConfig.disableSave;
+  const formContext = getFormContext({ form, user });
+  const pagesByChapter = createPageListByChapter(formConfig);
+  const chapters = chapterNames.reduce((chaptersAcc, chapterName) => {
+    const pages = pagesByChapter[chapterName];
+
+    const activePages = getActivePages(pages, formData);
+    const expandedPages = expandArrayPages(activePages, formData);
+    const chapterFormConfig = formConfig.chapters[chapterName];
+    const open = openChapters.includes(chapterName);
+    const pageKeys = getPageKeys(pages, formData);
+    const showUnviewedPageWarning = pageKeys.some(key => !viewedPages.has(key));
+
+    chaptersAcc.push({
+      expandedPages,
+      formConfig: chapterFormConfig,
+      name: chapterName,
+      open,
+      pageKeys,
+      showUnviewedPageWarning
+    });
```

Comment:
I like to avoid creating objects in the props but it's a good idea for arrays of views. This is probably the biggest part of the refactor because this data was in many places across three components. 

Consolidating this makes it easier to figure out what should go into the `viewState` and what can be memoized with selectors.

Reply:
I also think some of this naming could be tweaked. I would like it to be clearer what's going into the component renderer vs what is used for decisions. Right now it's just a

Reply:
I definitely like having all this in the mapStateToProps function. It makes all the transformation that's going on much clearer. I think we could also group some of this into reselect selectors to make it more efficient, since it looks like some of this calculation is unnecessary for every state change.